tests/test_quadruples_factory.py

```diff
@@ -0,0 +1,267 @@
+# -*- coding: utf-8 -*-
+
+"""Unit tests for quadruples factories."""
+
+import itertools as itt
+import unittest
+from typing import Collection, Optional
+from unittest.mock import patch
+
+import numpy as np
+import torch
+
+from pykeen.temporal.datasets import SmallSample
+from pykeen.quadruples import LCWAQuadrupleInstances, QuadruplesFactory
+from pykeen.quadruples.quadruples_factory import INVERSE_SUFFIX, QUADRUPLES_DF_COLUMNS
+
+quadruples = np.array(
+    [
+        ["peter", "likes", "chocolate_cake", "2019-01-01"],
+        ["chocolate_cake", "isA", "dish", "2021-03-01"],
+        ["susan", "likes", "pizza", "2023-09-26"],
+        ["peter", "likes", "susan", "1999-02-14"],
+    ],
+    dtype=str,
+)
+
+quadruples_with_nans = [
+    ["nan", "hates", "banana", "2001-08-29"],
+    ["susan", "nan", "dancing", "2010-09-10"],
+    ["apple", "isA", "nan", "2000-01-01"],
+    ["peter", "likes", "basketball", "nan"],
+]
+
+
+class TestQuadruplesFactory(unittest.TestCase):
+    """Class for testing quadruples factories."""
```

Comment:
when subclassing from TriplesFactory it makes sense to reduce to testing the functioality which is not already covered by TriplesFactory / or subclass from its test case.